packages/syft/src/syft/core/node/new/task/task_service.py

```diff
@@ -0,0 +1,400 @@
+                skip_permission=True,
+            )
+            if domain_input.is_ok():
+                inputs.update(domain_input.ok().base_dict[enclave_task.id])
+            else:
+                return domain_input.err()
+        return inputs
+
+    def execute_task(self, enclave_task: Task, context: AuthedServiceContext) -> None:
+        global stdout_
+        global stderr_
+
+        code = enclave_task.code
+        inputs = self.fetch_private_inputs(enclave_task=enclave_task, context=context)
+        outputs = enclave_task.outputs
+
+        try:
+            logger.info(f"inital outputs: {outputs}")
+
+            # Check overlap between inputs and vars
+            global_input_inter = set(globals().keys()).intersection(set(inputs.keys()))
+            local_input_inter = set(vars().keys()).intersection(set(inputs.keys()))
+
+            # If there's some intersection between global variables and input
+            if global_input_inter or local_input_inter:
+                stderr_message = " You can't use variable name: "
+                stderr_message += ",".join(list(global_input_inter))
+                stderr_message += ",".join(list(local_input_inter))
+
+                enclave_task.execution = "failed"
+                self.task_stash.update(enclave_task)
+                return Err("Variable conflicts in global space")
+
+            # create file-like string to capture ouputs
+            codeOut = StringIO()
+            codeErr = StringIO()
+
+            sys.stdout = codeOut
+            sys.stderr = codeErr
+
+            locals().update(inputs)
+            # byte_code = compile_restricted(code, "<string>", "exec")
+            # exec(byte_code, restricted_globals)
+            exec(code)  # nosec
+
+            for output in outputs:
+                logger.info(f"variable: {output} result: {vars()[output]}")
+                outputs[output] = vars()[output]
+
+            # restore stdout and stderr
+            sys.stdout = stdout_
+            sys.stderr = stderr_
+
+            logger.info(outputs)
+
+            logger.info("Error: " + str(codeErr.getvalue()))
+            logger.info("Std ouputs: " + str(codeOut.getvalue()))
+
+            new_id = UID()
+
+            dict_object = DictObject()
+            dict_object.base_dict = outputs
+            print("dict object", dict_object)
+            self.action_store.set(
+                uid=new_id,
+                credentials=context.credentials,
+                syft_object=dict_object,
+            )
+            print("Output id", new_id)
+            enclave_task.outputs = {"output_id": new_id}
+            enclave_task.execution = "Done"
+            self.task_stash.update(enclave_task)
+        except Exception as e:
+            sys.stdout = stdout_
+            sys.stderr = stderr_
+            raise e
+            print("Task Failed with Exception", e)
+        finally:
+            sys.stdout = stdout_
```

Comment:
MAYBE?: I think that having `finally` should also cover the `Exception` case (I think you can remove the `sys.stdout = stdout_` and `sys.stderr = stderr` from the `except Exception as e` case.

Reply:
One problem is the print statement would still use the replaced `sys.stdout/err`, but we can do `print("Task Failed with Exception", e, file=stderr_) # or stdout_` in addition to removing the duplicated lines.